### PR TITLE
Driver reorganization

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
Reopening this PR. Has updates to all drivers and in particular fleshes out the propulsion and docking drivers.

### CODE REVIEW

File: test/test_docking/test_docking.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#include <unity.h>
```
Comment (tanishqaggarwal) [inline test/test_docking/test_docking.cpp]: Will be populated in #58

File: test/test_prop/test_prop.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include <unity.h>
```
Comment (tanishqaggarwal) [inline test/test_prop/test_prop.cpp]: Will be populated in #58

File: test/test_docking/test_docking.cpp (hunk 1 above)
Comment (tanishqaggarwal) [reply test/test_docking/test_docking.cpp]: Populating this also depends on @fatimayousuf's updates to the Docking System driver in order to get it to be compatible with flight software.

File: lib/Drivers/DockingSystem.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    enable();
+
+    return true;
+}
+
+bool DockingSystem::is_functional() {
+    return digitalRead(DCDC::dcdc_sph_enable_pin)
+        && digitalRead(motor_sleep_pin);
+}
```
Comment (kkrol27) [inline lib/Drivers/DockingSystem.cpp]: I know this is all we can technically do, but just want to make sure we're all aware this isn't fool proof.

File: lib/Drivers/DockingSystem.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    unsigned int current_step = 0;
+
+    // step at a frequency of 500Hz
+    while (current_step < steps){
+        digitalWrite(motor_step_pin, LOW);
+        delayMicroseconds(1000);
+        digitalWrite(motor_step_pin, HIGH);
+        delayMicroseconds(1000);
+        current_step++;
+    }
```
Comment (kkrol27) [inline lib/Drivers/DockingSystem.cpp]: This is doing an individual step at each iteration triggered by a rising or falling edge correct?

Comment (tanishqaggarwal) [reply lib/Drivers/DockingSystem.cpp]: It's just briefly setting the motor PWM duty cycle to 100% and then to 0%.

This while loop is actually going to be removed on dev/docking_controller_2, in order to make the driver compatible with Flight Software.

**Pull request merged**